src/deepsparse/loggers/build_logger.py

```diff
@@ -325,6 +332,83 @@ def possibly_modify_target_identifiers(
+    accordingly
+
+    :param data_logging_config: The configuration of the data loggers
+    :return: The updated configuration of the data loggers
+    """
+    for target_identifier, metric_functions in data_logging_config.copy().items():
+        if target_identifier.endswith(PREDEFINED_IDENTIFIER_ALIAS):
+            # target_identifier is related to a predefined metric function
+            data_logging_config_from_predefined = predefined_metric_function(
+                metric_functions=metric_functions,
+                # subtracting -1 to account for `/` character
+                identifier_prefix=target_identifier[
+                    : -len(PREDEFINED_IDENTIFIER_ALIAS) - 1
+                ],
+            )
+            # update data_logging_config with the new
+            # data_logging_config_from_predefined
+            data_logging_config.update(data_logging_config_from_predefined)
+            # remove the old target_identifier key
+            del data_logging_config[target_identifier]
+    return data_logging_config
+
+
+def predefined_metric_function(
+    metric_functions: List[MetricFunctionConfig], identifier_prefix: str
+) -> Dict[str, List[MetricFunctionConfig]]:
+    """
+    Given a list of MetricFunctionConfig objects, parse out
+    the information about the pre-defined functions configuration.
+
+    Every MetricFunctionConfig.func in the `metric_functions` list
+    maps to a set of built-in functions and identifiers. Thus,
+    the list of metric function configuration effectively is transformed
+    in a data logging configuration
+
+    :param metric_functions: A list containing MetricFunctionConfig
+        objects that specify the predefined data logging configuration.
+    :return: Data logging configuration from the predefined metric functions
+    """
+    new_data_logging_config = {}
+    for metric_function in metric_functions:
+        function_group_name = metric_function.func
+        # fetch the pre-defined data logging configuration from the registry
+        registered_function_group = DATA_LOGGING_REGISTRY.get(function_group_name)
+        if not registered_function_group:
+            raise ValueError(
+                f"Unknown function group name: {function_group_name}. "
+                f"Supported function group names: {list(DATA_LOGGING_REGISTRY.keys())}"
+            )
+        for (
+            registered_identifier,
+            registered_functions,
+        ) in registered_function_group.items():
+            target_identifier = get_target_identifier(
+                target_name=registered_identifier, pipeline_identifier=identifier_prefix
+            )
+            new_metric_functions = [
+                MetricFunctionConfig(
+                    func=func,
+                    frequency=metric_function.frequency,
+                    target_loggers=metric_function.target_loggers,
+                )
+                for func in registered_functions
+            ]
```

Comment:
why allocate a new list instead of append to the config? (could use defaultdict)